tony-core/src/main/java/com/linkedin/tony/TFConfig.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * TFConfig POJO for serialization.
+ */
+public class TFConfig {
```

Comment:
I don't think we can do this, currently the ObjectMapper uses the getters to construct the json object (e.g. getCluster -> "cluster", getTask -> "task") so we don't want to mix frameworks, otherwise we'll have extraneous stuff in the json object

Reply:
The cluster info class should be generic, all distributed training framework would require a similar information (role/index/total_num_of_tasks). TF/PyTorch should derive their json from this.

Reply:
Seems like PyTorch doesn't require the entire cluster spec to be passed to all processes. PyTorch has 3 methods of initialization:

* provide TCP address of rank 0 process
* set some ENVIRONMENT VARIABLES describing rank 0 host/port, number of processes, rank of current process
* path to non-existent file on a shared filesystem - seems like first process to create it is rank 0 and other processes' ranks are assigned automatically

Ref: https://pytorch.org/docs/stable/distributed.html#initialization

Reply:
Yeah IMO it's hard to make this kind of information fully generic...we have no idea what information other frameworks will require/not require. In any case I see your point @oliverhu, but perhaps we can address this as other framework support is added.

Reply:
The TCP address of rank 0 process is like the cluster spec.. get WORKER INDEX 0 of this cluster spec.

What I'm picturing this is like a generalized information of the whole application cluster. The information can be leveraged for further customized communication between nodes as well.

Reply:
@erwa

Reply:
@hungj ok, we can do that later. As I mentioned earlier, not only for frameworks but also for TaskExecutors - for example for the workers node to do some extra features.